src/packageurl/contrib/url2purl.py

```diff
@@ -431,54 +455,54 @@ def build_github_api_purl(url):
 # https://codeload.github.com/nexB/scancode-toolkit/tar.gz/v3.1.1
 # https://codeload.github.com/berngp/grails-rest/zip/release/0.7
 github_codeload_pattern = (
-    r"https?://codeload.github.com/(?P<namespace>.+)/(?P<name>.+)/"
-    r"(zip|tar.gz|tar.bz2|tgz)/(.*/)*"
+    r"https?://codeload\.github\.com/(?P<namespace>.+)/(?P<name>.+)/"
+    r"(zip|tar\.gz|tar\.bz2|tgz)/([^/]*/)*"
```

Comment:
Clarification for this change from `(.*/)*` to `([^/]*/)*` can be found here:

https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS